### DISCUSSION TITLE
Genmon goes away...

### DISCUSSION THREAD

**Discussion by alabamatoy** [jgyates/genmon#1092] [Questions]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).

### Version

Genmon 1.19.02
OS Ubuntu 22.04 LTS

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have not submitted logs

### What is/are your question(s)?

Genmon "goes away".  I dont know what else to call it, but the web interface simply stops responding.  I cannot find any errors in syslog or the genmon or genserv logs.  The following brings it back again:
```
> bash /genmon/./startgenmon.sh start
Starting genmon python scripts
Starting....
Stopping genmon.py
Starting /genmon/genmon.py 
Stopping genserv.py
Starting /genmon/genserv.py 
```
This indicates to me that startgenmon believes that both genmon.py and genserv.py are still running.  But the web interface is non-responsive.  It seems to happen every 24-36 hours or so.  How do I troubleshoot this?  And, yes, I know, I can cron a restart, but that seems like treating the symptom, not the underlying disease.  Perhaps if startgenmon.sh had a "status" function?

**Comment by jgyates** [comment]
First, I would submit your log files as I have no data formulate any suggestions.

Second, instead is starting genmon and genserv again (without stopping it first), I would run this command:

  ./startgenmon.sh restart 

The log files report errors and show when the programs start.

In almost all cases that I have seen similar to this have been traced back to local network issues. You can search past issues on the "Router" tag to see these issues. Some routers will block local web servers once detected. If this is the case for you then you will either need to whitelist the genmon server, or disable this feature in your router.

I am responding on a mobile device at the moment however I will try to send more info on troubleshooting later tonight. The ps and grep commands can be used to show genmon is running. Also the top program will show active processes

**Reply by alabamatoy** [reply]
I submitted logs via the about page.  Dont know if it succeeded or not, got no discernible response to hitting the submit button.

"In almost all cases that I have seen similar to this have been traced back to local network issues. ... Some routers will block local web servers once detected. "  I dont think this is my issue.  The only way I access the Pi is via Webmin, and access to Webmin has not been an issue.

**Reply by jgyates** [reply]
I have not received logs. Outbound email must be enabled to send logs.

Restarting vis the scrip will print a warning if the process is not active.

Are you exposing gemon to an external WAN port? 

Genserv.py is the web server and genmon communicates with the controller and exports data to display. You can validate that genmon is running and functional by using the ClientInterface.py program. The wiki has details on this but it allows you to display the text info exported by genmon without the web interface. This will allow you to isolate the issue to genserv.py rather than genmon.py.

**Reply by alabamatoy** [reply]
"I have not received logs. Outbound email must be enabled to send logs.", Ah, well, that's not working so....

**Reply by alabamatoy** [reply]
"Are you exposing gemon to an external WAN port?" Only port 443.

**Reply by jgyates** [reply]
That is likely the problem. The flask webserver will was not designed for heavy traffic and the port is likely being probed. There have been instances of brute force attacks resulting in unwanted generator starting by the attacker. It is not recommended you expose any ports that genmon is using. It it is likely that genserv experiencing a brute force attack and the web server is failing. If this is the case then they will eventually succeed. The wiki known issues page has more info. The recommended remote access method is to use a VPN. Also, you can search past issues for "VPN" and find the brute force incidents.

**Reply by alabamatoy** [reply]
OK.  Port is closed.  We'll see.....

**Comment by jgyates** [comment]
Also, are you starting genmon on boot via crontab?

**Reply by alabamatoy** [reply]
Yes, but the system has not been rebooted in 13 days  I thought reboots on Linux were for hardware upgrades? :-)   The croned command is 
`sleep 30 && /bin/bash /genmon/startgenmon.sh start`
The next time the web interface goes away, I will check running processes to see if its still there before restarting genmon.  I bet its turning zombie somehow.

**Comment by jgyates** [comment]
Here is more info from past issues that are similar to your:

https://github.com/jgyates/genmon/discussions/1029
https://github.com/jgyates/genmon/issues/919
https://github.com/jgyates/genmon/issues/892
https://github.com/jgyates/genmon/issues/271
